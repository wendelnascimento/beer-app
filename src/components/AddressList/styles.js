import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const List = styled.ul`
  list-style-type: none;
  padding: 16px 0;
  margin: 0;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  border-top: 1px solid #eee;
  box-shadow: rgba(0, 0, 0, 0.6) 0px 14px 23px -16px;
  background: #ffffff;
  max-height: 300px;
  overflow-y: scroll;
`;

export const ListItem = styled.li`
  padding: 8px 24px;
  width: 100%;

  &:not(:last-child) {
    border-bottom: 1px solid #fafafa;
    padding-bottom: 24px;
  }
`;

export const ItemTitle = styled.span`
  display: block;
  font-size: 1.2rem;
  color: #000000;
`;

export const ItemVicinity = styled.span`
  display: block;
  color: #cccccc;
  font-size: 1rem;
`;

export const CustomLink = styled(Link)`
  text-decoration: none;
`;
