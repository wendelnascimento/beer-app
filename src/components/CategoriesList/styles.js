import styled from 'styled-components';

export const List = styled.ul`
  list-style-type: none;
  padding: 0;
  margin: 0 0 16px;
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-gap: 16px;

  @media screen and (max-width: 900px) {
    grid-template-columns: repeat(5, 1fr);
  }

  @media screen and (max-width: 600px) {
    grid-template-columns: repeat(4, 1fr);
  }

  @media screen and (max-width: 460px) {
    grid-template-columns: repeat(3, 1fr);
  }

  @media screen and (max-width: 350px) {
    grid-template-columns: repeat(2, 1fr);
  }
`;

export const ListItem = styled.li`
  border: 2px solid #000;
  border-radius: 5px;
  overflow: hidden;
  transition: border-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);

  &:hover {
    border-color: #f28e1c;
  }
`;

export const ItemButton = styled.button`
  width: 100%;
  padding: 8px 16px;
  border: none;
  font-weight: 700;
  text-align: center;
  background: #ffffff;

  &:hover {
    cursor: pointer;
  }
`;
