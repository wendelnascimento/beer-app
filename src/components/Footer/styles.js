import styled from 'styled-components';

export const Wrapper = styled.footer`
  padding: 16px;
  color: #cccccc;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  border-top: 1px solid #fafafa;
`;
