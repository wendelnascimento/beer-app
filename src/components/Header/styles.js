import styled from 'styled-components';

export const Wrapper = styled.header`
  max-width: 100vw;
  background: #fff;
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border-bottom: 1px solid #fafafa;
`;

export const Title = styled.h1`
  font-weight: 300;
  margin: 0;
`;
