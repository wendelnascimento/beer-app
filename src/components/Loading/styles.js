import styled, { keyframes } from 'styled-components';

export const bounceAndRotate = keyframes`
  0% {
    transform: scale3d(.8, .8, 1);
  }
  25% {
    transform: scale3d(1.5, 1.5, 1);
  }
  50% {
    transform: scale3d(.8, .8, 1);
  }
  75% {
    transform: rotate3d(0, 0, 1, 180deg);
  }
  100% {
    transform: scale3d(1, 1, 1), rotate3d(0, 0, 1, 360deg);
  }
`;

export const Wrapper = styled.div`
  position: fixed;
  z-index: 10;
  width: 100vw;
  height: 100vh;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgba(255, 255, 255, 0.6);
`;

export const Beer = styled.span`
  font-size: 7rem;
  transform: scale3d(1, 1, 1), rotate3d(0, 0, 1, 0deg);
  animation: ${bounceAndRotate} 2s infinite linear;
`;
