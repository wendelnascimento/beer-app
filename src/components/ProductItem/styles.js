import styled from 'styled-components';

export const Card = styled.li`
  width: 100%;
  max-width: 300px;
  background: #ffffff;
  border: 2px solid #fff;
  display: flex;
  flex-direction: column;
  border-radius: 5px;
  box-shadow: rgba(0, 0, 0, 0.6) 0px 2px 15px -5px;
  transform: translate3d(0, 0, 0);
  transition: transform 0.3s cubic-bezier(0.19, 1, 0.22, 1),
    border-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);
  overflow: hidden;

  &:hover {
    transform: translate3d(0, -5px, 0);
    border-color: #f28e1c;
  }
`;

export const ProductImage = styled.img`
  width: 100%;
  object-fit: scale-down;
  background: #ffffff;
  border-radius: 5px;
  margin-bottom: 8px;
`;

export const ImagePlaceholder = styled.span`
  font-size: 5rem;
  text-align: center;
  margin-top: auto;
  margin-bottom: auto;
`;

export const ProductTitle = styled.span`
  color: #666;
  font-size: 0.8rem;
  margin-top: auto;
  padding: 0 8px;
`;

export const ProductPrice = styled.span`
  color: #f28e1c;
  font-size: 1.2rem;
  font-weight: 700;
  margin-bottom: 16px;
  padding: 0 8px;
`;

export const QuantityRow = styled.div`
  align-self: stretch;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const AddButton = styled.button`
  width: 30px;
  height: 30px;
  background: #f28e1c;
  color: #ffffff;
  border: none;
  font-size: 1.3rem;
  font-weight: 300;

  &:hover {
    cursor: pointer;
  }
`;

export const RemoveButton = styled(AddButton)`
  background: #b03c3c;
`;
