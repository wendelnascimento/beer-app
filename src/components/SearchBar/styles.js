import styled from 'styled-components';

export const CustomInput = styled.input`
  padding: 24px;
  border-radius: 5px;
  border: 2px solid #fff;
  box-shadow: rgba(0, 0, 0, 0.6) 0px 2px 15px -5px;
  width: 70%;
  font-size: 1.5rem;
  outline: none;
  transition: border 0.3s cubic-bezier(0.19, 1, 0.22, 1);

  &::placeholder {
    color: #ddd;
  }

  &:hover,
  &:focus {
    border: 2px solid #f28e1c;
  }
`;
