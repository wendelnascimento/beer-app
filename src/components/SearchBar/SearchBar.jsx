import React from 'react';
import PropTypes from 'prop-types';

import { CustomInput, InputWrapper, ErrorMessage } from './styles';

const SearchBar = ({
  onChange, value, loading, error, placeholder, errorMessage,
}) => (
  <div>
    <InputWrapper loading={loading.toString()}>
      <CustomInput
        type="search"
        placeholder={placeholder}
        onChange={onChange}
        value={value}
        error={error.toString()}
        aria-invalid={error}
        aria-errormessage="errorMessageSearchBar"
      />
    </InputWrapper>
    {error && <ErrorMessage id="errorMessageSearchBar">{errorMessage}</ErrorMessage>}
  </div>
);

SearchBar.defaultProps = {
  loading: false,
  error: false,
  placeholder: 'Ex: Rua Américo Brasiliense, São Paulo',
  errorMessage: 'Houve um problema ao encontrar o endereço solicitado, tente novamente.',
};

SearchBar.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  loading: PropTypes.bool,
  error: PropTypes.bool,
  placeholder: PropTypes.string,
  errorMessage: PropTypes.string,
};

export default SearchBar;
