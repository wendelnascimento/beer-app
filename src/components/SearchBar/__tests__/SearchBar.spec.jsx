import React from 'react';
import { shallow, mount } from 'enzyme';
import { css } from 'styled-components';

import SearchBar from '../SearchBar';
import { InputWrapper, loadingAnimation } from '../styles';

describe('<SearchBar />', () => {
  const props = {
    onChange: jest.fn(),
    value: '',
  };

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('does render properly', () => {
    shallow(<SearchBar onChange={props.onChange} value={props.value} />);
  });

  it('matches stored snapshot', () => {
    const wrapper = mount(<SearchBar onChange={props.onChange} value={props.value} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('does have correct placeholder', () => {
    const wrapper = mount(<SearchBar onChange={props.onChange} value={props.value} />);

    expect(wrapper.find('input').props().placeholder).toEqual(
      'Ex: Rua Américo Brasiliense, São Paulo',
    );
  });

  it('does call onChange when we type on it', () => {
    const wrapper = mount(<SearchBar onChange={props.onChange} value={props.value} />);

    wrapper.find('input').simulate('change', { target: { value: 'Test' } });

    expect(props.onChange).toHaveBeenCalledTimes(1);
  });

  it('does update the value accordingly to prop change', () => {
    const wrapper = mount(<SearchBar onChange={props.onChange} value={props.value} />);

    wrapper.setProps({
      value: 'Test',
    });

    expect(wrapper.find('input').getDOMNode().value).toEqual('Test');
  });

  it('renders loading state', () => {
    const wrapper = mount(<SearchBar onChange={props.onChange} value={props.value} loading />);

    /* The two lines below are a hack, because if you wrap the entire line into an css`` rule
       you'll gonna see an error from styled components, because the keyframes will stil be an
       object. So I just got the generated name to simulate the rule.
    */
    const keyframesObject = css`
      ${loadingAnimation}
    `;
    const animation = `${keyframesObject[1].name} 3s infinite ease-in-out alternate`;

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(InputWrapper)).toHaveStyleRule('animation', animation, {
      modifier: ':after',
    });
  });
});
