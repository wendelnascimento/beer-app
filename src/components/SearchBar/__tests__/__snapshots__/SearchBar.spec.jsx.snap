// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> matches stored snapshot 1`] = `
.c0 {
  width: 100%;
  position: relative;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  box-shadow: rgba(0,0,0,0.6) 0px 2px 15px -5px;
  overflow-x: hidden;
}

.c0:after {
  content: '';
  position: absolute;
  bottom: 0;
  left: -100%;
  height: 2px;
  width: 100%;
  background: #000000;
  -webkit-transform: translate3d(0%,0,0);
  -ms-transform: translate3d(0%,0,0);
  transform: translate3d(0%,0,0);
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-animation: 3s infinite ease-in-out alternate;
  animation: 3s infinite ease-in-out alternate;
}

.c1 {
  padding: 24px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  border: 2px solid #fff;
  width: 100%;
  font-size: 1.5rem;
  outline: none;
  -webkit-transition: border 0.3s cubic-bezier(0.19,1,0.22,1);
  transition: border 0.3s cubic-bezier(0.19,1,0.22,1);
  -webkit-animation: 2s 4 cubic-bezier(0.19,1,0.22,1);
  animation: 2s 4 cubic-bezier(0.19,1,0.22,1);
}

.c1::-webkit-input-placeholder {
  color: #ddd;
}

.c1::-moz-placeholder {
  color: #ddd;
}

.c1:-ms-input-placeholder {
  color: #ddd;
}

.c1::placeholder {
  color: #ddd;
}

.c1:hover,
.c1:focus {
  border: 2px solid #f28e1c;
}

<SearchBar
  error={false}
  errorMessage="Houve um problema ao encontrar o endereço solicitado, tente novamente."
  loading={false}
  onChange={[MockFunction]}
  placeholder="Ex: Rua Américo Brasiliense, São Paulo"
  value=""
>
  <div>
    <styled.div
      loading="false"
    >
      <div
        className="c0"
        loading="false"
      >
        <styled.input
          aria-errormessage="errorMessageSearchBar"
          aria-invalid={false}
          error="false"
          onChange={[MockFunction]}
          placeholder="Ex: Rua Américo Brasiliense, São Paulo"
          type="search"
          value=""
        >
          <input
            aria-errormessage="errorMessageSearchBar"
            aria-invalid={false}
            className="c1"
            onChange={[MockFunction]}
            placeholder="Ex: Rua Américo Brasiliense, São Paulo"
            type="search"
            value=""
          />
        </styled.input>
      </div>
    </styled.div>
  </div>
</SearchBar>
`;

exports[`<SearchBar /> renders error state 1`] = `
.c0 {
  width: 100%;
  position: relative;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  box-shadow: rgba(0,0,0,0.6) 0px 2px 15px -5px;
  overflow-x: hidden;
}

.c0:after {
  content: '';
  position: absolute;
  bottom: 0;
  left: -100%;
  height: 2px;
  width: 100%;
  background: #000000;
  -webkit-transform: translate3d(0%,0,0);
  -ms-transform: translate3d(0%,0,0);
  transform: translate3d(0%,0,0);
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-animation: 3s infinite ease-in-out alternate;
  animation: 3s infinite ease-in-out alternate;
}

.c1 {
  padding: 24px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  border: 2px solid #fff;
  width: 100%;
  font-size: 1.5rem;
  outline: none;
  -webkit-transition: border 0.3s cubic-bezier(0.19,1,0.22,1);
  transition: border 0.3s cubic-bezier(0.19,1,0.22,1);
  -webkit-animation: dbNLkw 2s 4 cubic-bezier(0.19,1,0.22,1);
  animation: dbNLkw 2s 4 cubic-bezier(0.19,1,0.22,1);
}

.c1::-webkit-input-placeholder {
  color: #ddd;
}

.c1::-moz-placeholder {
  color: #ddd;
}

.c1:-ms-input-placeholder {
  color: #ddd;
}

.c1::placeholder {
  color: #ddd;
}

.c1:hover,
.c1:focus {
  border: 2px solid #f28e1c;
}

.c2 {
  display: block;
  color: #b03c3c;
  font-size: 0.8rem;
  margin-top: 8px;
  opacity: 0;
  -webkit-animation: jnNqVJ 2s forwards cubic-bezier(0.19,1,0.22,1);
  animation: jnNqVJ 2s forwards cubic-bezier(0.19,1,0.22,1);
}

<SearchBar
  error={true}
  errorMessage="Houve um problema ao encontrar o endereço solicitado, tente novamente."
  loading={false}
  onChange={[MockFunction]}
  placeholder="Ex: Rua Américo Brasiliense, São Paulo"
  value=""
>
  <div>
    <styled.div
      loading="false"
    >
      <div
        className="c0"
        loading="false"
      >
        <styled.input
          aria-errormessage="errorMessageSearchBar"
          aria-invalid={true}
          error="true"
          onChange={[MockFunction]}
          placeholder="Ex: Rua Américo Brasiliense, São Paulo"
          type="search"
          value=""
        >
          <input
            aria-errormessage="errorMessageSearchBar"
            aria-invalid={true}
            className="c1"
            onChange={[MockFunction]}
            placeholder="Ex: Rua Américo Brasiliense, São Paulo"
            type="search"
            value=""
          />
        </styled.input>
      </div>
    </styled.div>
    <styled.strong
      id="errorMessageSearchBar"
    >
      <strong
        className="c2"
        id="errorMessageSearchBar"
      >
        Houve um problema ao encontrar o endereço solicitado, tente novamente.
      </strong>
    </styled.strong>
  </div>
</SearchBar>
`;

exports[`<SearchBar /> renders loading state 1`] = `
.c0 {
  width: 100%;
  position: relative;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  box-shadow: rgba(0,0,0,0.6) 0px 2px 15px -5px;
  overflow-x: hidden;
}

.c0:after {
  content: '';
  position: absolute;
  bottom: 0;
  left: -100%;
  height: 2px;
  width: 100%;
  background: #000000;
  -webkit-transform: translate3d(0%,0,0);
  -ms-transform: translate3d(0%,0,0);
  transform: translate3d(0%,0,0);
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-animation: dgfBtW 3s infinite ease-in-out alternate;
  animation: dgfBtW 3s infinite ease-in-out alternate;
}

.c1 {
  padding: 24px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  border: 2px solid #fff;
  width: 100%;
  font-size: 1.5rem;
  outline: none;
  -webkit-transition: border 0.3s cubic-bezier(0.19,1,0.22,1);
  transition: border 0.3s cubic-bezier(0.19,1,0.22,1);
  -webkit-animation: 2s 4 cubic-bezier(0.19,1,0.22,1);
  animation: 2s 4 cubic-bezier(0.19,1,0.22,1);
}

.c1::-webkit-input-placeholder {
  color: #ddd;
}

.c1::-moz-placeholder {
  color: #ddd;
}

.c1:-ms-input-placeholder {
  color: #ddd;
}

.c1::placeholder {
  color: #ddd;
}

.c1:hover,
.c1:focus {
  border: 2px solid #f28e1c;
}

<SearchBar
  error={false}
  errorMessage="Houve um problema ao encontrar o endereço solicitado, tente novamente."
  loading={true}
  onChange={[MockFunction]}
  placeholder="Ex: Rua Américo Brasiliense, São Paulo"
  value=""
>
  <div>
    <styled.div
      loading="true"
    >
      <div
        className="c0"
        loading="true"
      >
        <styled.input
          aria-errormessage="errorMessageSearchBar"
          aria-invalid={false}
          error="false"
          onChange={[MockFunction]}
          placeholder="Ex: Rua Américo Brasiliense, São Paulo"
          type="search"
          value=""
        >
          <input
            aria-errormessage="errorMessageSearchBar"
            aria-invalid={false}
            className="c1"
            onChange={[MockFunction]}
            placeholder="Ex: Rua Américo Brasiliense, São Paulo"
            type="search"
            value=""
          />
        </styled.input>
      </div>
    </styled.div>
  </div>
</SearchBar>
`;
